integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/client/TopicReceiver.java

```diff
@@ -0,0 +1,142 @@
+     * @throws JMSException
+     */
+    public TopicReceiver(String topicName, JMSAcknowledgeMode acknowledgeMode,
+            ConfigurationReader configurationReader, String durableSubId) throws IOException, NamingException,
+            JMSException {
+
+        // map of config key and config value
+        Map<String, String> clientConfigPropertiesMap = configurationReader.getClientConfigProperties();
+        Properties properties = new Properties();
+        properties.put(Context.INITIAL_CONTEXT_FACTORY, ClientConstants.FILE_INITIAL_CONNECTION_FACTORY);
+        properties.put(ClientConstants.CONNECTION_FACTORY_NAME_PREFIX + ClientConstants.ANDES_CONNECTION_FACTORY_NAME,
+                ClientUtils.getTCPConnectionURL(clientConfigPropertiesMap));
+        properties.put(ClientConstants.TOPIC_NAME_PREFIX + topicName, topicName);
+        InitialContext ctx = new InitialContext(properties);
+
+        TopicConnectionFactory connectionFactory = (TopicConnectionFactory) ctx
+                .lookup(ClientConstants.ANDES_CONNECTION_FACTORY_NAME);
+        topicConnection = connectionFactory.createTopicConnection();
+        topicConnection.start();
+        topicSession = topicConnection.createTopicSession(false, acknowledgeMode.getType());
+        Topic topic = (Topic) ctx.lookup(topicName);
+        if (durableSubId != null) {
+            subscriber = topicSession.createDurableSubscriber(topic, durableSubId);
+        } else {
+            subscriber = topicSession.createSubscriber(topic);
+        }
+    }
+
+    /**
+     * Close JMS connection/session and clean up resources.
+     *
+     * @throws JMSException
+     */
+    public void closeReceiver() throws JMSException {
+        try {
+            subscriber.close();
+            topicSession.close();
+            topicConnection.stop();
+            topicConnection.close();
+        } catch (NullPointerException e) {
+            //ignore
+        }
+
+    }
+
+    /**
+     * get Maximum Messages Count.
+     */
+    public int getMaximumMessageCount() {
+        return maximumMessageCount;
+    }
+
+    /**
+     * set Maximum Messages Count.
+     */
+    public void setMaximumMessageCount(int maximumMessageCount) {
+        this.maximumMessageCount = maximumMessageCount;
+    }
+
+    /**
+     * Receive a message from a topic in given timeout.
+     * @param timeout Maximum time to wait for the message
+     * @return Received message
+     * @throws JMSException
+     */
+    public Message receiveMessage(long timeout) throws JMSException {
+        return subscriber.receive(timeout);
+    }
+
+    /**
+     * Get all the received messages count by looping until a timeout occurred.
+     * @return Received messages count
+     * @throws JMSException
+     */
+    public int getMessageCount() throws JMSException {
+        int count = 0;
+        while (true) {
+            Message message = subscriber.receive(ClientConstants.CLIENT_WAIT_DURATION);
```

Comment:
The CLIENT_WAIT_DURATION is 1s. Will that be enough?

Reply:
The tests are passing in local machine.

Reply:
Since this is a timeout I'll increase to 10s.